### PR TITLE
Adapt ODS-488,1303,771 to account for dashboard differences in Self-managed

### PR DESCRIPTION
**Pull request by bdattoma**
From v1.18, if RHODS is installed as Self-managed, then dashboard should hide some of the ISVs (https://issues.redhat.com/browse/RHODS-4907):
- starburst galaxy
- rhoam
- Nvidia GPU addon

This PR makes ODS-488 changing the expected data based on RHODS installation type.

For convenience, the PR changes also ODS-1303 and ODS-771 which are affected by Self-managed installation

Signed-off-by: bdattoma [bdattoma@redhat.com](mailto:bdattoma@redhat.com)

### CODE REVIEW

File: tests/Resources/Common.robot [hunk 1]
```diff
@@ -156,6 +156,16 @@ Skip If RHODS Version Greater Or Equal Than
...
     END
 
+Skip If RHODS Is Self-managed
+    [Documentation]    Skips test if RHODS is installed as Self-managed
+    [Arguments]    ${msg}=${EMPTY}
+    ${is_self_managed}=    Is RHODS Self-Managed
+    IF    "${msg}" != "${EMPTY}"
+       Skip If    condition=${is_self_managed}==True    msg=${msg}
+    ELSE
+       Skip If    condition=${is_self_managed}==True    msg=This test is skipped for Self-managed RHODS
```
Comment (lugi0) [inline tests/Resources/Common.robot:166]: Is this needed? Can't you just pass an empty msg as a parameter?

**Review by lugi0 — APPROVED** [review]
lgtm, one small doubt

File: tests/Resources/Common.robot (hunk 1 above)
Comment (bdattoma) [reply tests/Resources/Common.robot:166]: I copied this implementation from `Skip If RHODS Version Greater Or Equal Than` so I thought it was already analyzed as option, and I left as it is

File: tests/Resources/Common.robot [hunk 2]
```diff
@@ -156,6 +156,16 @@ Skip If RHODS Version Greater Or Equal Than
        Skip If    condition=${version-check}==True    msg=This test is skipped for RHODS ${version} or greater
     END
 
+Skip If RHODS Is Self-managed
```
Comment (jgarciao) [inline tests/Resources/Common.robot:159]: Does robocop allow a dash in the keyword name?  
Please, consider also capitalize Managed, same as in `Is RHODS Self-Managed`

File: tests/Tests/100__deploy/100__installation/101__installation.robot [hunk 3]
```diff
@@ -33,6 +33,7 @@ Verify User Can Access RHODS Documentation From OCM Console
   [Tags]  ODS-1303
   ...     Tier2
   ...     AutomationBug
```
Comment (jgarciao) [inline tests/Tests/100__deploy/100__installation/101__installation.robot:35]: Is `AutomationBug` still valid?

File: tests/Resources/OCP.resource [hunk 4]
```diff
@@ -1,6 +1,7 @@
 *** Settings ***
 Documentation  Set of Keywords for OCP checks
 Library    OpenShiftLibrary
+Library    ../../venv_ods/lib64/python3.8/site-packages/robot/libraries/OperatingSystem.py
```
Comment (jgarciao) [inline tests/Resources/OCP.resource:4]: I think this should be removed

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: tests/Resources/Common.robot (hunk 2 above)
Comment (bdattoma) [reply tests/Resources/Common.robot]: it doesn't complain about the dash. I capitalized it

File: tests/Tests/100__deploy/100__installation/101__installation.robot (hunk 3 above)
Comment (bdattoma) [reply tests/Tests/100__deploy/100__installation/101__installation.robot]: hm I think it's safe to remove the label, I see it succeeded last run. Previous failures were labeld as "system issue"

File: tests/Resources/OCP.resource (hunk 4 above)
Comment (bdattoma) [reply tests/Resources/OCP.resource]: done

**Comment by bdattoma** [comment]
Tested.

ODS-488: 1 (expected) failure
image
The only failure present is due to a wrong RHOSAK link reported in Explore section of RHODS Dashboard. Already asked to fix it

ODS-771: successfully skipped one of the checks if RHODS is Self-Managed
image

ODS-1303: successfully skipped the test if RHODS is Self-Managed
image

**Review by jgarciao — APPROVED** [review]
lgtm

**Review by lugi0 — APPROVED** [review]

**Pull request merged**